spec/jobs/import_data_from_rapids_job_spec.rb

```diff
@@ -186,17 +176,137 @@
+          }.to change(OccupationStandard, :count).by(1)
+
+          occupation_standard = OccupationStandard.last
+
+          expect(occupation_standard.redacted_document).to_not be_attached
+        end
+      end
+
+      context "when document is mark as uploaded" do
+        context "when response has a document" do
+          it "attaches the document" do
+            stub_get_token!
+            create_registration_agency_for("MI", agency_type: :oa)
+
+            occupation_standard_response = create_list(
+              :rapids_api_occupation_standard,
+              1,
+              :competency,
+              :with_wps_document,
+              wpsDocument: "https://entbpmpstg.dol.gov/suite/webapi/rapids/data-sharing/documents/wps/123456"
+            )
+            rapids_response = create(:rapids_response, totalCount: 1, wps: occupation_standard_response)
+
+            stub_rapids_api_response(
+              {
+                batchSize: ImportDataFromRAPIDSJob::PER_PAGE_SIZE,
+                startIndex: 1
+              },
+              rapids_response
+            )
+
+            document = File.read(Rails.root.join(
+              "spec", "fixtures", "files", "document.docx"
+            ))
+
+            stub_documents_response("123456", document)
+
+            expect {
+              described_class.perform_now
+            }.to change(OccupationStandard, :count).by(1)
+
+            occupation_standard = OccupationStandard.last
+
+            expect(occupation_standard.redacted_document).to be_attached
+          end
+        end
+
+        context "when response does not have a document" do
+          it "skips document attachment" do
+            stub_get_token!
+            create_registration_agency_for("MI", agency_type: :oa)
+
+            occupation_standard_response = create_list(
+              :rapids_api_occupation_standard,
+              1,
+              :competency,
+              :with_wps_document,
+              wpsDocument: "https://entbpmpstg.dol.gov/suite/webapi/rapids/data-sharing/documents/wps/123456"
+            )
+            rapids_response = create(:rapids_response, totalCount: 1, wps: occupation_standard_response)
+
+            stub_rapids_api_response(
+              {
+                batchSize: ImportDataFromRAPIDSJob::PER_PAGE_SIZE,
+                startIndex: 1
+              },
+              rapids_response
+            )
+
+            stub_documents_response("123456", nil)
+
+            expect {
+              described_class.perform_now
+            }.to change(OccupationStandard, :count).by(1)
+
+            occupation_standard = OccupationStandard.last
+
+            expect(occupation_standard.redacted_document).to_not be_attached
+          end
+        end
+      end
+    end
   end
 end
 
 def stub_get_token!
   allow_any_instance_of(RAPIDS::API).to receive(:get_token!).and_return "xxx"
 end
 
+def create_registration_agency_for(state_abbreviation, agency_type:)
```

Comment:
Is it worthwhile to push this into factory_bot, e.g.

```ruby
FactoryBot.define do
  factory :registration_agency do
    traits_for_enum :agency_type, RegistrationAgency.agency_types

    ### remove `state` association, build it manually instead
    transient do
      state_abbreviation { nil }
    end

    agency_type { :oa }

    trait :for_national_program do
      state { nil }
    end

    after(:build) do |registration_agency, context|
      state_attribs = {}

      if context.state_abbreviation
        state_attribs[:abbreviation] = context.state_abbreviation
      end

      registration_agency.state = FactoryBot.build(:state, **state_attribs)
    end
  end
end
```

Then we can do

```ruby
create(:registration_agency, state_abbreviation: "MI", agency_type: :oa)
```

(I haven't tried this code, it might not work.)

Reply:
@mike-burns I like that! I'll create another PR with the suggestion. Thanks!